File: Dijikstra Two Stack/TwoStack.js
let twoStack = function(stringInput)
{
    let array = stringInput.split("");
    let functions = ["%","+","*","/","-"];
    let operators = [];
    let values = [];
    
    for(let i = 0; i<array.length;i++)
    {   
        let s = array[i];
        if(s=="(") {}
        else if(functions.includes(s) == true)
        {
            operators.push(s);
        }
        else if(s==")")
        {
            let op = operators.pop();
            if(op == "+") {values.push(values.pop() + values.pop())}
            else if(op == "*") {values.push(values.pop() * values.pop())}
            else if(op == "-") {values.push(values.pop() - values.pop())}
            else if(op == "/") {values.push(values.pop() / values.pop())}
            else if(op == "%") {values.push(values.pop() % values.pop())}
        }
        else
        {
            values.push(parseFloat(s));
        };

    };
    return values.pop();
};


let stringInput = "(1+((2+3)*(4*5)))";
let result = twoStack(stringInput);
console.log(result);

File: Dijikstra Two Stack/TwoStack.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Fri May  3 19:15:11 2019

@author: musahni
"""
    
    

def twoStack(stringInput):
    stringList = list(stringInput)
    functions = list("*/+-%")
    operators = []
    values = []
    for s in stringList:
        if s == "(":
            pass
        elif s in functions:
            operators.append(s)
        elif s == ")":
            op = operators.pop()
            if op == "+":
                values.append(values.pop() + values.pop())
            elif op == "*":
                values.append(values.pop() * values.pop())
            elif op == "-":
                values.append(values.pop() - values.pop())
            elif op == "/":
                values.append(values.pop() / values.pop())
            elif op == "%":
                values.append(values.pop() % values.pop())
        else:
            values.append(float(s))
    return values.pop()
            
g = twoStack("(1+((2+3)*(4*5)))")


